.github/actions/context.js

```diff
@@ -0,0 +1,95 @@
+// @ts-check
+
+/**
+ * Extracts inputs from context based on event name and properties.
+ * run_id is only defined for "workflow_run:completed" events.
+ * 
+ * @param {import('github-script').AsyncFunctionArguments['github']} github
+ * @param {import('github-script').AsyncFunctionArguments['context']} context
+ * @param {import('github-script').AsyncFunctionArguments['core']} core
+ * @returns {Promise<{owner: string, repo: string, head_sha: string, issue_number: number, run_id: number }>}
+ */
+async function extractInputs(github, context, core) {
+  core.info(`extractInputs(${context.eventName}, ${context.payload.action})`);
+
+  // Add support for more event types as needed
+  if (context.eventName === "pull_request") {
+    const payload =
+      /** @type {import("@octokit/webhooks-types").PullRequestEvent} */ (
+        context.payload
+      );
+
+    const inputs = {
+      owner: payload.repository.owner.login,
+      repo: payload.repository.name,
+      head_sha: payload.pull_request.head.sha,
+      issue_number: payload.number,
+      run_id: NaN
+    };
+  
+    core.info(`inputs: ${JSON.stringify(inputs)}`);
+  
+    return inputs;
+  } else if (
+    context.eventName === "workflow_run" &&
+    context.payload.action === "completed"
+  ) {
+    const payload =
+      /** @type {import("@octokit/webhooks-types").WorkflowRunCompletedEvent} */ (
+        context.payload
+      );
+
+    let issue_number;
+
+    const pull_requests = payload.workflow_run.pull_requests;
+    if (pull_requests && pull_requests.length > 0) {
+      // For non-fork PRs, we should be able to extract the PR number from the payload, which avoids an
+      // unnecessary API call.  The listPullRequestsAssociatedWithCommit() API also seems to return
+      // empty for non-fork PRs.
+      issue_number = pull_requests[0].number;
+    } else {
+      // For fork PRs, we must call an API in the head repository to get the PR number in the target repository
```

Comment:
Wow that is unfortunate that they don't even give you the PR number back for forked builds. That seems like an oversight of some sort.

Reply:
https://securitylab.github.com/resources/github-actions-preventing-pwn-requests/

Search for "workflow_run context".